### PR TITLE
[fix] Fix simulator compilation warnings

### PR DESCRIPTION
**Pull request by enkiller**
## 拉取/合并请求描述：(PR description)

[
修复 VS 模拟器编译警告
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [ ] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

File: bsp/simulator/rtconfig_project.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 #ifdef _MSC_VER
 /* disable some warning in MSC */
+#pragma warning(disable:4103)   /* structure packing changed by including file */
```
Comment (mysterywolf) [inline bsp/simulator/rtconfig_project.h:17]:
```suggestion
// #pragma warning(disable:4103)   /* structure packing changed by including file */
```
建议改成这种方式，可以提示但是不要直接disable掉，因为这个并没有实际解决问题，只是在自欺欺人。

File: components/dfs/filesystems/romfs/dfs_romfs.c [hunk 2]
```diff
@@ -157,7 +157,7 @@ int dfs_romfs_read(struct dfs_fd *file, void *buf, size_t count)
 
 int dfs_romfs_lseek(struct dfs_fd *file, off_t offset)
 {
-    if (offset <= file->size)
```
Comment (mysterywolf) [inline components/dfs/filesystems/romfs/dfs_romfs.c:160]: 建议用rt_size_t

File: components/dfs/filesystems/romfs/dfs_romfs.c [hunk 3]
```diff
@@ -261,7 +262,7 @@ int dfs_romfs_getdents(struct dfs_fd *file, struct dirent *dirp, uint32_t count)
         return -EINVAL;
 
     index = 0;
-    for (index = 0; index < count && file->pos < file->size; index ++)
+    for (index = 0; index < count && (size_t)file->pos < file->size; index ++)
```
Comment (mysterywolf) [inline components/dfs/filesystems/romfs/dfs_romfs.c:265]: rt_size_t

File: components/libc/compilers/common/ctime.c [hunk 4]
```diff
@@ -186,11 +186,11 @@ struct tm *gmtime_r(const time_t *timep, struct tm *r)
...
     r->tm_min = work % 60;
     r->tm_hour = work / 60;
-    work = *timep / (SPD);
+    work = (int)(*timep / (SPD));
     r->tm_wday = (4 + work) % 7;
     for (i = 1970;; ++i)
     {
-        time_t k = __isleap(i) ? 366 : 365;
+        int k = __isleap(i) ? 366 : 365;
         if (work >= k)
```
Comment (mysterywolf) [inline components/libc/compilers/common/ctime.c:194]: 这块把time_t改成int也是因为给出警告了嘛

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: components/dfs/filesystems/romfs/dfs_romfs.c [hunk 5]
```diff
@@ -157,7 +157,7 @@ int dfs_romfs_read(struct dfs_fd *file, void *buf, size_t count)
 
 int dfs_romfs_lseek(struct dfs_fd *file, off_t offset)
 {
-    if (offset <= file->size)
+    if (offset >= 0 && (size_t)offset <= file->size)
```
Comment (mysterywolf) [inline components/dfs/filesystems/romfs/dfs_romfs.c:160]:
```suggestion
    if (offset >= 0 && (rt_size_t)offset <= file->size)
```

File: components/dfs/filesystems/romfs/dfs_romfs.c (hunk 3 above)
Comment (mysterywolf) [inline components/dfs/filesystems/romfs/dfs_romfs.c:265]:
```suggestion
    for (index = 0; index < count && (rt_size_t)file->pos < file->size; index ++)
```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by mysterywolf — APPROVED** [review]

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**